src/decoder.rs

```diff
@@ -492,7 +520,8 @@ impl<R: Read> Decoder<R> {
                     if let Some(data) = parse_app(&mut self.reader, marker)? {
                         match data {
                             AppData::Adobe(color_transform) => {
-                                self.color_transform = Some(color_transform)
+                                self.color_transform =
+                                    ColorTransform::AdobeColorTransform(color_transform)
```

Comment:
This seems like it leads to confusing semantics. If one explicitly sets up `ColorTransform::None` but then it's overwritten by an app segment? Which semantics did you intend, could you document them on `set_transform` as well?

Reply:
Hmm this is a good point. What do you think to:

```rust
/// Sets the colour transform to be applied before returning the data. If must_use is set to true, 
/// then this transform will be used for decoding, and app segments relating to colour transforms 
/// will be ignored. If must_use is set to false, then this transform will be used only in the case 
/// where no app segment overwrites it.
    pub fn set_color_transform(&mut self, transform: ColorTransform, must_use: bool)
```

Reply:
Not sure, I'm not too fond of `bool` parameters but would maybe use a separate function for 'hints' vs. overrides, for instance because not all variants may be applicable as hints. Also, a hint might have the semantics of falling back to the default behavior in case `CMYK` is not applicable where an override errors in 3-component color.

Choices, so many choices :smile:

Reply:
Not quite sure if I understand what you mean with 'hints', but we could do:

```rust
/// Colour transform to use if one is not set using `set_color_transform` or found in app segments.
pub fn set_fallback_color_transform(&mut self, transform: ColorTransform) {
...
}

/// Colour transform to use when decoding the image. App segments relating to colour transforms 
/// will be ignored.
pub fn set_color_transform(&mut self, transform: ColorTransform) {
...
}
```

Alternatively, we could split out the fallback based on number of channels, like so:

```rust
/// Colour transform to use when the data has 3 channels and if one is not set using `set_color_transform` or found in app segments.
pub fn set_fallback_color_transform_3_channels(&mut self, transform: ColorTransform) {
...
}

```rust
/// Colour transform to use when the data has 4 channels and if one is not set using `set_color_transform` or found in app segments.
pub fn set_fallback_color_transform_4_channels(&mut self, transform: ColorTransform) {
...
}

```

But those are some long function names....

Reply:
Option 1 actually looks good.

Reply:
Yeah. Perhaps even `set_fallback_color_transform` vs. `override_color_transform`.